src/main/java/ch/njol/skript/expressions/ExprSignText.java

```diff
@@ -32,145 +26,203 @@
+		if (delta == null) {
+			if (mode != ChangeMode.DELETE && mode != ChangeMode.RESET)
+				return;
+		}
+		Block block = exprBlock.getSingle(event);
+		if (block == null)
 			return;
-		if (getTime() >= 0 && e instanceof SignChangeEvent && b.equals(((SignChangeEvent) e).getBlock()) && !Delay.isDelayed(e)) {
+		if (isLines) {
+			if (!sign.isOfType(block))
+				return;
+			Sign state = (Sign) block.getState();
 			switch (mode) {
+				case ADD:
+					for (String text : (String[]) delta) {
+						int index = 0;
+						while (!state.getLine(index).equals("")) {
+							index++;
+							if (index > 3)
+								break;
+						}
+						if (index <= 3)
+							state.setLine(index, Utils.replaceChatStyles(text));
+					}
+					break;
 				case DELETE:
-					((SignChangeEvent) e).setLine(line, "");
+					for (int index = 0; index < 4; index++) {
+						state.setLine(index, "");
+					}
 					break;
-				case SET:
-					assert delta != null;
-					((SignChangeEvent) e).setLine(line, (String) delta[0]);
+				case REMOVE:
+					for (String text : (String[]) delta) {
+						for (int index = 0; index <= 3; index++) {
+							if (state.getLine(index).equals(Utils.replaceChatStyles(text))) {
+								state.setLine(index, "");
+								break;
+							}
+						}
+					}
 					break;
-			}
-		} else {
-			if (!sign.isOfType(b))
-				return;
-			final Sign s = (Sign) b.getState();
-			switch (mode) {
-				case DELETE:
-					s.setLine(line, "");
+				case REMOVE_ALL:
+					for (String text : (String[]) delta) {
+						for (int index = 0; index <= 3; index++) {
+							if (state.getLine(index).equals(Utils.replaceChatStyles(text))) {
+								state.setLine(index, "");
+							}
+						}
+					}
 					break;
 				case SET:
-					assert delta != null;
-					s.setLine(line, (String) delta[0]);
+					for (int index = 0; index < Math.min(4, delta.length); index++) {
+						state.setLine(index, Utils.replaceChatStyles((String) delta[index]));
+					}
 					break;
 			}
-			if (hasUpdateBooleanBoolean) {
-				try {
-					s.update(false, false);
-				} catch (final NoSuchMethodError err) {
-					hasUpdateBooleanBoolean = false;
-					s.update();
-				}
+			state.update(false, false);
+		} else {
+			String text = "";
+			if (mode == ChangeMode.SET)
+				text = Utils.replaceChatStyles((String) delta[0]);
+			Number lineNumber = line.getSingle(event);
+			if (lineNumber == null || lineNumber.intValue() < 1 || lineNumber.intValue() > 4)
+				return;
+			int index = lineNumber.intValue() - 1;
+			if (event instanceof SignChangeEvent && getTime() >= 0 && block.equals(((SignChangeEvent) event).getBlock()) && !delayed) {
+				((SignChangeEvent) event).setLine(index, text);
 			} else {
-				s.update();
+				if (!sign.isOfType(block)) {
+					return;
+				}
+				Sign state = (Sign) block.getState();
+				switch (mode) {
+					case SET:
+						state.setLine(index, text);
+						break;
+					case DELETE:
+						state.setLine(index, "");
+						break;
+				}
```

Comment:
This switch statement only reflects two changemodes, also there is no defiance of the other changemodes or even default. This switch statement serves no purpose being a switch statement. use conditions instead or instill the entire else statement in a switch statement.